devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -0,0 +1,530 @@
+        capability "Light"
+        capability "Configuration"
+        capability "Refresh"
+        capability "Health Check"
+
+        attribute "firmwareVersion", "string"
+        attribute "lastCheckIn", "string"
+        attribute "syncStatus", "string"
+
+        fingerprint mfr: "0312", prod: "C000", model: "C009", deviceJoinName: "Minoston Mini Plug" // old MP21Z
+        fingerprint mfr: "0312", prod: "FF00", model: "FF0C", deviceJoinName: "Minoston Mini Plug" //MP21Z Minoston Mini Smart Plug
+    }
+
+    simulator { }
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.Lighting.light13", backgroundColor:"#00a0dc", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.Lighting.light13", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'TURNING ON', action:"switch.off", icon:"st.lights.philips.hue-single", backgroundColor:"#00a0dc", nextState:"turningOff"
+				attributeState "turningOff", label:'TURNING OFF', action:"switch.on", icon:"st.lights.philips.hue-single", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+		}
+		standardTile("refresh", "device.refresh", width: 2, height: 2) {
+			state "refresh", label:'Refresh', action: "refresh"
+		}
+		valueTile("syncStatus", "device.syncStatus", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "syncStatus", label:'${currentValue}'
+		}
+		standardTile("sync", "device.configure", width: 2, height: 2) {
+			state "default", label: 'Sync', action: "configure"
+		}
+		valueTile("firmwareVersion", "device.firmwareVersion", decoration:"flat", width:3, height: 1) {
+			state "firmwareVersion", label:'Firmware ${currentValue}'
+		}
+		main "switch"
+		details(["switch", "refresh", "syncStatus", "sync", "firmwareVersion"])
+	}
+    preferences {
+        configParams.each {
+            createEnumInput("configParam${it.num}", "${it.name}:", it.value, it.options)
+        }
+    }
+}
+
+private createEnumInput(name, title, defaultVal, options) {
+    input name, "enum",
+            title: title,
+            required: false,
+            defaultValue: defaultVal.toString(),
+            options: options
+}
+
+def installed() {
+    logDebug "installed()..."
+
+}
+
+def updated() {
+    if (!isDuplicateCommand(state.lastUpdated, 5000)) {
+        state.lastUpdated = new Date().time
+
+        logDebug "updated()..."
+
+        initialize()
+
+        runIn(5, executeConfigureCmds, [overwrite: true])
+    }
+    return []
+}
+
+private initialize() {
+    def checkInterval = ((60 * 60 * 3) + (5 * 60))
+
+    def checkIntervalEvt = [name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"]]
```

Comment:
Please remove `offlinePingable: "1"`, as this option is reserved for more crucial devices.

Reply:
OK, I remove them!